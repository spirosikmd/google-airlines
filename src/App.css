/*
 * Colors from https://coolors.co/fefffe-e5fcf5-b3dec1-210124-750d37
 * - #FEFFFE
 * - #E5FCF5
 * - #B3DEC1
 * - #210124
 * - #750D37
 */

.app {
  padding: 0 25px 25px;
}

.app__title {
  font-weight: 500;
}

.app__info {
  color: hsl(0, 0%, 40%);
  line-height: 25px;
  max-width: 60ch;
}

.app__airlines {
  padding-top: 25px;
}

.app__airline-list {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(75px, 75px));
  grid-gap: 25px;
}

.app__airline {
  padding: 20px;
  border: 1px solid #e5fcf5;
  border-radius: 50%;
  transition: box-shadow 0.2s ease-out;
  display: flex;
  justify-content: center;
  align-items: center;
}

.app__airline:hover,
.app__airline:focus {
  box-shadow: 0 0 3px #750d37;
}

.app__anchor {
  color: #750d37;
}

.app__tooltip {
  transition: opacity 0.2s ease-out;
}
