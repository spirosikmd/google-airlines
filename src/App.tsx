import React from 'react';
import ReactTooltip from 'react-tooltip';
import AirlineList from './AirlineList';
import Info from './Info';
import './App.css';
import { AirlineData } from './airlines';

const App = ({ airlines }: { airlines: AirlineData[] }) => (
  <div className="app">
    <Info />
    <AirlineList airlines={airlines} />
    <ReactTooltip place="top" effect="solid" />
  </div>
);

export default App;
