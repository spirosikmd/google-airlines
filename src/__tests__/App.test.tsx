import React from 'react';
import { render, cleanup } from 'react-testing-library';
import App from '../App';

afterEach(cleanup);

const airlines = [
  { id: 'iata1_airline1', iata: 'iata1', airline: 'airline1' },
  { id: 'iata2_airline2', iata: 'iata2', airline: 'airline2' }
];

test('renders the app with info and a list of airlines', () => {
  const { asFragment } = render(<App airlines={airlines} />);

  expect(asFragment()).toMatchSnapshot();
});
