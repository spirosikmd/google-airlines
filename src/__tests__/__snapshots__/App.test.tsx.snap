// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the app with info and a list of airlines 1`] = `
<DocumentFragment>
  <div
    class="font-sans p-6"
  >
    <h1
      class="text-4xl font-medium"
    >
      Google Airlines
    </h1>
    <p
      class="max-w-2xl text-gray-600"
    >
      Google created their own airline logos for 
      <a
        class="text-purple-600 underline cursor-pointer"
        href="https://www.google.com/flights"
        target="_blank"
      >
        google flights
      </a>
       for many airlines out there. This is a collection of a few of them in a random order. I got the IATA codes from 
      <a
        class="text-purple-600 underline cursor-pointer"
        href="https://en.wikipedia.org/wiki/List_of_airline_codes"
        target="_blank"
      >
        wikipedia
      </a>
      . Enjoy!
    </p>
    <div
      class="pt-6"
    >
      <div
        class="flex flex-wrap"
      >
        <div
          class="p-6 m-2 border border-solid rounded-full border-gray-200  hover:border-purple-600"
          currentitem="false"
          data-tip="airline1"
          tabindex="0"
        >
          <img
            alt="airline1"
            class="h-8 w-8"
            src="data:image/svg+xml;charset=utf-8,%3Csvg xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg' width%3D'200' height%3D'150' viewBox%3D'0 0 200 150'%2F%3E"
            style="background: rgb(229, 252, 245);"
          />
        </div>
        <div
          class="p-6 m-2 border border-solid rounded-full border-gray-200  hover:border-purple-600"
          currentitem="false"
          data-tip="airline2"
          tabindex="0"
        >
          <img
            alt="airline2"
            class="h-8 w-8"
            src="data:image/svg+xml;charset=utf-8,%3Csvg xmlns%3D'http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg' width%3D'200' height%3D'150' viewBox%3D'0 0 200 150'%2F%3E"
            style="background: rgb(229, 252, 245);"
          />
        </div>
      </div>
    </div>
    <div
      class="__react_component_tooltip place-top type-dark "
      data-id="tooltip"
    />
  </div>
</DocumentFragment>
`;
